python/spacewalk/server/importlib/backend.py

```diff
@@ -3295,10 +3292,11 @@ def _do_diff(self, data, table_name, uq_fields, fields):
     def validate_pks(self):
         # If nevra is enabled use checksum as primary key
         tbs = self.tables["rhnPackage"]
-        if not CFG.ENABLE_NVREA:
-            # remove checksum from a primary key if nevra is disabled.
-            if "checksum_id" in tbs.pk:
-                tbs.pk.remove("checksum_id")
+        with cfg_component("server") as cfg:
+            if not cfg.ENABLE_NVREA:
```

Comment:
For me looks like `AttributeError: ENABLE_NVREA` is still possible here if it's missing in rhn.conf or default configs for some reason.

Reply:
We have it in our default `rhn_server.conf`, if there is an `AttributeError` there is a bigger problem (default configs missing or modified), IMO it's okay to show a backtrace in that situation.